src/imgui/Shortcuts.cc

```diff
@@ -0,0 +1,164 @@
+#include "Shortcuts.hh"
+#include "imgui_internal.h"
+
+#include "SettingsConfig.hh"
+
+#include <array>
+
+namespace openmsx {
+
+//-- (Only) the table in this section will be edited when adding new shortcuts
+
+struct AllShortcutInfo {
+    ShortcutIndex index;
+    ImGuiKeyChord keyChord;
+    ShortcutType type;
+    bool repeat;
+    std::string_view name; // used in settings.xml
+    std::string_view description; // shown in GUI
+};
+
+// When adding a new Shortcut, we only:
+//  * Add a new value in the above 'enum ShortcutIndex'
+//  * Add a single line in this table
+// Note: if you look at the generated code on the right, this table does NOT end up in the generated code
+//       instead it's only used to calculate various other tables (that do end up in the generated code)
+//       those 'calculations' are all done at compile-time
+// Note: the first column is ONLY used to verify (at-compile-time) whether the rows in this
+//       table are in the same order as the values in ShortcutIndex
+static constexpr auto allShortcutInfo = std::to_array<AllShortcutInfo>({
+//   Index           keychord                    global repeat name                         description
+    {GOTO_MEM,       ImGuiMod_Ctrl | ImGuiKey_G, ShortcutType::LOCAL,  false, "goto_memory_address",       "Go to address in hex viewer"},
+    {STEP,           ImGuiKey_F6,                ShortcutType::GLOBAL, true,  "step",                      "Single step in debugger"},
+    {BREAK,          ImGuiKey_F7,                ShortcutType::GLOBAL, false, "break",                     "Break CPU emulation in debugger"},
+    {GOTO_DISASM,    ImGuiMod_Ctrl | ImGuiKey_G, ShortcutType::LOCAL,  false, "goto_disasm_address",       "Go to address in dissassembler"},
+    {MOVE_UP_MEM,    ImGuiKey_UpArrow,           ShortcutType::LOCAL,  true,  "move_up_memory_address",    "Move up in hex viewer"},
+    {MOVE_DOWN_MEM,  ImGuiKey_DownArrow,         ShortcutType::LOCAL,  true,  "move_down_memory_address",  "Move down in hex viewer"},
+    {MOVE_LEFT_MEM,  ImGuiKey_LeftArrow,         ShortcutType::LOCAL,  true,  "move_left_memory_address",  "Move left in hex viewer"},
+    {MOVE_RIGHT_MEM, ImGuiKey_RightArrow,        ShortcutType::LOCAL,  true,  "move_right_memory_address", "Move right hex viewer"},
+});
```

Comment:
Not required for this PR. But we must address this before the release:

The `name` fields from this table get saved in `settings.xml`, and thus for backwards compatibility we can't easily change these names. So it's important that we pick good names from the start.

I don't like the names that have "memory" in them. Because they don't only apply to the "memory hex editor", but to all hex-editors. Maybe names like `hex_editor_goto_address` and `hex_editor_move_up` are better?